journalbeat/docs/general-options.asciidoc

```diff
@@ -0,0 +1,66 @@
+[[configuration-general-options]]
+== Specify general settings
+
+You can specify settings in the +{beatname_lc}.yml+ config file to control the
+general behavior of {beatname_uc}. This includes:
+
+* <<configuration-global-options,Global options>> that control things like
+publisher behavior and the location of some files.
+
+* <<configuration-general,General options>> that are supported by all Elastic
+Beats.
+
+[float]
+[[configuration-global-options]]
+=== Global {beatname_uc} configuration options
+
+These options are in the +{beatname_lc}+ namespace.
+
+[float]
+[id="{beatname_lc}-registry-file"]
+==== `registry_file`
+
+The name of the registry file. If a relative path is used, it is considered relative to the
+data path. See the <<directory-layout>> section for details. The default is `${path.data}/registry`.
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc}.registry_file: registry
+----
+
+[float]
+[id="{beatname_lc}-backoff"]
+==== `backoff`
+
+The number of seconds to wait before trying to read again from journals. The
+default is 1s.
```

Comment:
why is backoff global?

Reply:
@urso That's how it is in the reference.yml. :-) @kvch Is this correct?

Reply:
@dedemorton checking the reference config file it's a per input setting

@kvch  do we even need a backoff for reading journals? Don't we get a 'signal' when there are new logs?

Reply:
@urso The sdjournal API provides an `sd_journal_wait` function which polls the journal and returns an event after a configured timeout. Possible events returned are journal is unchanged, new entry in journal or error. Either way I had to poll something to learn if there is a new entry. Thus, I decided to poll from Journalbeat instead of sdjournal API to save a few calls to C. If I used `sd_journal_wait` I had to call `sd_journal_next` to get the new entry. Now I simply call `sd_journal_next` which return `nil` if there is no new entry. Also, this let me to reuse the existing `common.backoff` functionality of libbeat.

Reply:
@urso But `backoff` and `max_backoff` also show up under global options here:

https://github.com/elastic/beats/blob/master/journalbeat/journalbeat.yml#L47

@kvch Is that a mistake in the conf.yaml?

Reply:
I moved the options to the input section for now. I noticed that `backoff_factor` is no longer shown in the reference.yml, so I removed it.

Reply:
`backoff`, `max_backoff`, `include_matches` and `seek` are removed from global options. `backoff_factor` was removed a few weeks ago. Seems like I haven't updated the GH issue with the config, sorry.

@urso I have decreased `max_backoff` to 20 seconds. WDYT?

Reply:
> I have decreased max_backoff to 20 seconds. WDYT?

SGTM

Reply:
@kvch 60s is the correct default for 6.5, though, correct?